src/Product.jl

```diff
@@ -0,0 +1,94 @@
+@doc doc"""
+    Product{TM<:Tuple, TRanges<:Tuple, TSizes<:Tuple} <: Manifold
+
+Product manifold $M_1 \times M_2 \times \dots \times M_n$ with product geometry.
+`TRanges` and `TSizes` statically define the relationship between representation
+of the product manifold and representations of point, tangent vectors
+and cotangent vectors of respective manifolds.
+
+# Constructor
+
+    Product(M_1, M_2, ..., M_n)
+
+generates the product manifold $M_1 \times M_2 \times \dots \times M_n$
+"""
+struct Product{TM<:Tuple, TRanges, TSizes} <: Manifold
+    manifolds::TM
+end
+
+function Product(manifolds...)
+    sizes = map(m -> representation_size(m, MPoint), manifolds)
+    lengths = map(s -> prod(s), sizes)
```

Comment:
Could do `map(prod, sizes)`

Reply:
Why wouldn't this implementation of `representation_size` work when a product of more than two manifolds is represented?

Thanks for the tips about `mapreduce` and `prod`. :+1: